staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -149,19 +149,30 @@ func (g *genValidations) GenerateType(c *generator.Context, t *types.Type, w io.
 }
 
 func (g *genValidations) hasValidations(n *typeNode) bool {
+	seen := map[*typeNode]bool{}
+	return g.hasValidationsImpl(n, seen)
+}
+
+func (g *genValidations) hasValidationsImpl(n *typeNode, seen map[*typeNode]bool) bool {
+	if seen[n] {
+		return false
```

Comment:
nit: given there is a hasValidationsCache method it might makes sense to have some comment explaining the seen map here is for cycle detection - eg: 
```go
if seen[n] {
  // return when re-visiting a node to prevent infinite recursion
  return false
}
```

Reply:
ACK, good call